### PR TITLE
PERF-1463 Optional OAuth client secret

### PR DESCRIPTION
**Pull request by troex**
Some Bing/Azure apps require "client secret" - especially apps created earlier in MS live dev console, passing client secret it optional and shouldn't affect apps created without secret.

This is proper and complete implementation of #9

### CODE REVIEW

File: tasks/bing_ads_ruby_sdk.rake [hunk 1]
```diff
@@ -2,17 +2,19 @@ require 'dotenv/load'
 
 namespace :bing_token do
   desc "Gets and stores Bing OAuth token in file"
-  task :get, [:filename, :bing_developer_token, :bing_client_id] do |task, args|
+  task :get, [:filename, :bing_developer_token, :bing_client_id, :bing_client_secret] do |task, args|
 
     filename = args[:filename] || ENV.fetch('BING_STORE_FILENAME')
     developer_token = args[:bing_developer_token] || ENV.fetch('BING_DEVELOPER_TOKEN')
     bing_client_id = args[:bing_client_id] || ENV.fetch('BING_CLIENT_ID')
+    bing_client_secret = args[:bing_client_secret] || ENV.fetch('BING_CLIENT_SECRET')
```
Comment (danielmbarlow) [inline tasks/bing_ads_ruby_sdk.rake]: Won't this require that the `bing_client_secret` be provided as an argument or an ENV var to the Rake task?

If you use one of these the Rake task arg will be optional:
* `ENV.fetch('BING_CLIENT_SECRET', nil)`
* `ENV['BING_CLIENT_SECRET']`

File: lib/bing_ads_ruby_sdk/oauth2/authorization_handler.rb [hunk 2]
```diff
@@ -73,15 +73,15 @@ def extract_codes(url)
...
+      def build_client(developer_token, client_id, client_secret=nil)
         Signet::OAuth2::Client.new({
           authorization_uri:    'https://login.microsoftonline.com/common/oauth2/v2.0/authorize',
           token_credential_uri: 'https://login.microsoftonline.com/common/oauth2/v2.0/token',
           redirect_uri:         'https://login.microsoftonline.com/common/oauth2/nativeclient',
           developer_token: developer_token,
           client_id: client_id,
           scope: 'offline_access'
-        })
+        }.merge(client_secret ? { client_secret: client_secret } : {}))
```
Comment (apneadiving) [inline lib/bing_ads_ruby_sdk/oauth2/authorization_handler.rb]:
```suggestion
        }.tap do |hash|
          hash[:client_secret] = client_secret if client_secret
        end
```

File: tasks/bing_ads_ruby_sdk.rake (hunk 1 above)
Comment (apneadiving) [inline tasks/bing_ads_ruby_sdk.rake]:
```suggestion
    bing_client_secret = args[:bing_client_secret] || ENV.fetch('BING_CLIENT_SECRET', nil)
```

**Comment by apneadiving** [comment]
gonna be addressed in new version. Thanks for reporting and sorry for delay :) https://github.com/Effilab/bing_ads_ruby_sdk/pull/13

**Pull request closed without merging**